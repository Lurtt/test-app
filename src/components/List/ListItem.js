import React from 'react'
import styled from 'styled-components'

import { Label } from '../'

const ListItemStyle = styled.div`
  display: flex;
  width: 100%;
  margin-bottom: 1rem;
  border: 1px solid #ccc;
  box-shadow: 2px 3px 4px rgba(0, 0, 0, 0.1);
`

const Field = styled.div`
  flex: 1;
  padding: 1rem;
`

export default function(props) {
  return (
    <ListItemStyle>
      <Field>
        <Label>File name</Label>
        <div>{props.name}</div>
      </Field>
      <Field>
        <Label>Date creation</Label>
        <div>{new Date(props.creationDate).toString().slice(3, 15)}</div>
      </Field>
      <Field>
        <Label>Status</Label>
        <div>{props.status}</div>
      </Field>
    </ListItemStyle>
  )
}
