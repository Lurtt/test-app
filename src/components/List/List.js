import React from 'react'
import styled from 'styled-components'

import ListItem from './ListItem'

const ListStyle = styled.section`
  display: flex;
  flex-direction: column;
  align-items: center;
`

export default function(props) {
  return (
    <ListStyle>
      {props.data.map(item => <ListItem key={item.id} {...item} />)}
    </ListStyle>
  )
}
