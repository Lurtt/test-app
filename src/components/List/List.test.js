import React from 'react'
import renderer from 'react-test-renderer'

import List from './List'

jest.mock('./ListItem', () => 'ListItem')

describe('<List />', () => {
  test('Should render', () => {
    const mockData = []
    const result = renderer.create(<List data={mockData} />).toJSON()
    expect(result).toMatchSnapshot()
  })

  test('Should render items', () => {
    const mockData = [
      {
        id: 1,
        creationDate: '2007-05-16',
        name: 'Commodo.mp3',
        status: 'In Process',
      },
    ]
    const result = renderer.create(<List data={mockData} />).toJSON()
    expect(result).toMatchSnapshot()
  })
})
