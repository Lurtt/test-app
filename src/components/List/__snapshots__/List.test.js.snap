// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> Should render 1`] = `
<section
  className="sc-bdVaJa bxcuYP"
/>
`;

exports[`<List /> Should render items 1`] = `
<section
  className="sc-bdVaJa bxcuYP"
>
  <ListItem
    creationDate="2007-05-16"
    id={1}
    name="Commodo.mp3"
    status="In Process"
  />
</section>
`;
